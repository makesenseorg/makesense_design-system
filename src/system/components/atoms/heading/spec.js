import { shallowMount } from "@vue/test-utils";
import Comp from "./Heading.vue";

describe("Heading.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        tag: "h1",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("sets class with tagClass modifier", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        tag: "h1",
        tagClass: "h2",
      },
    });
    expect(wrapper.classes()).toContain("heading--h2");
  });

  it("has a default color class", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        tag: "h1",
      },
    });
    expect(wrapper.classes()).toContain("heading--text");
  });

  it("sets class with color modifier", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        tag: "h1",
        color: "secondary",
      },
    });
    expect(wrapper.classes()).toContain("heading--secondary");
  });
});
