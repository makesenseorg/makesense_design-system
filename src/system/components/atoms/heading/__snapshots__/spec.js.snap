// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading.vue matches snapshot 1`] = `
<h1
  class="heading heading--text heading--undefined"
/>
`;
